packages/fast-check/package.json

```diff
@@ -8,17 +8,26 @@
     "./package.json": "./package.json",
     ".": {
       "require": {
+        "types@<5.7": "./lib/cjs/types57/fast-check.d.ts",
         "types": "./lib/cjs/types/fast-check.d.ts",
         "default": "./lib/cjs/fast-check.js"
       },
       "import": {
+        "types@<5.7": "./lib/types57/fast-check.d.ts",
```

Comment:
I believe we need both `typesVersions` and `types` on `exports`: https://www.typescriptlang.org/docs/handbook/release-notes/typescript-4-9.html#exports-is-prioritized-over-typesversions